### ISSUE TITLE
.NET nanoFramework extension broken on 17.3

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/project-system#8518]
_This issue has a corresponding [ticket on Developer Community](https://developercommunity.visualstudio.com/t/NET-nanoFramework-extension-broken-on-1/10116629). Please vote and comment there to make sure your voice is heard._

---
After updating VS to 17.3 .NET nanoFramework extension is broken.
When opening a nfproj get this output:

```
LimitedFunctionality
System.ArgumentNullException: Value cannot be null.
Parameter name: commandLine
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioProjectOptionsProcessor.SetCommandLine(String commandLine)
   at Microsoft.VisualStudio.ProjectSystem.LanguageServices.WorkspaceProjectContextProvider.d7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.ProjectSystem.LanguageServices.WorkspaceProjectContextProvider.d7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.Rethrow(Exception ex)
   at Microsoft.VisualStudio.ProjectSystem.ExceptionFilter.<>cDisplayClass2_0.gAction|0()
   at GuardMethodClass.GuardMethod(Func1 , Func2 , Func`2 )
```

This is using the latest extension here
https://github.com/nanoframework/nf-Visual-Studio-extension/releases/tag/v2022.2.0.31

The same extension for VS2019 is working just fine.

This is, most likely, something missing in our project system. Possibly in a targets file, just that we don't have enough details or visibility in what's happening inside VS.

---
### Original Comments

_This section cut as all comments were private; please see the original feedback ticket for more details._

**Comment by tmeschter** [comment]
Short explanation: the nanoFramework needed to define the `CommandLineArgsForDesignTimeEvaluation` property in their .props/.targets. We were trying to read this property and pass it along to the language service, and threw an exception when it wasn't present.

The nanoFramework added the property and is now unblocked. This issue tracks reporting a better exception message that includes the name of the missing property. This may help third parties, such as nanoFramework, identify and fix issues on their own, and will _definitely_ help us if/when similar issues come our way.

**Comment by tmeschter** [comment]
Note that since the original feedback ticket was filed the relevant code paths have been significantly updated, and we're more resilient to a missing `CommandLineArgsForDesignTimeEvaluation`.

**Comment by raducoriu** [comment]
The error came back on VS 17.5.0

**Comment by drewnoakes** [comment]
I believe it's a subtly different issue and should be fixed in 17.5.1. Could you please verify?

**Comment by josesimoes** [comment]
I've just updated VS to 17.5.1 and I do NOT have any issues opening and building C# nanoFramework projects.
